### DISCUSSION TITLE
413 Error on Form Uploads over 6MB

### DISCUSSION THREAD

**Discussion by StokleyO** [sveltejs/kit#10777] [Q&A] [answered]
Hey everyone!

Recently I've been getting a 413 Error on a form of mine, any time I upload anything over 6MB. Additionally, this only happens in Production, and not on my Local Machine. Would anyone happen to know why that is? If it helps, I'm deploying using Netlify, my database is hosted on Vercel, and I'm using AWS to handle the file upload. If anyone could help out, it would be greatly appreciated

**Comment by StokleyO** [comment]
Has anyone ever had this issue? It's weird that it only happens in Production

**Comment by teemingc** [comment, marked as answer]
The Netlify lambda function can only process a limited body size. 

> 6 MB request and response payload size limit for buffered synchronous functions

https://docs.netlify.com/functions/overview/#default-deployment-options

You'll need to upload the file directly to AWS without passing the file through the serverless function.

https://docs.aws.amazon.com/AmazonS3/latest/userguide/PresignedUrlUploadObject.html

**Reply by StokleyO** [reply]
@s3812497 do you happen to have an example of what that might look like? The context here is that users are uploading audio through this form, so I'm not sure how they could still do that on my site if it can't go through the function

**Reply by teemingc** [reply]
https://dev.to/kitsunekyo/upload-to-aws-s3-directly-from-the-browser-js-aws-sdk-v3-1opk

This blog post illustrates the flow quite well (although it's not specific to SvelteKit, but is similar enough).

**Reply by StokleyO** [reply]
Thank you @s3812497 you are insanely helpful!

**Comment by StokleyO** [comment]
Thank you for taking a look @s3812497, on the Netlify forums they were adamant that they don't surface a 413 error so this must be a coding issue. I had come across that same statement in the documentation. I really appreciate your help
